lib/tests/triples.nix

```diff
@@ -0,0 +1,157 @@
+# Run:
+# [nixpkgs]$ nix-build -o expected-output lib/tests/triples.nix -A expected-output-from-gnu-config
+# [nixpkgs]$ nix-instantiate --eval --strict -E '(import lib/tests/triples.nix).tests ./expected-output'
+# Expected output: [], or the failed cases
+# Two steps are required to avoid IFD.
+#
+# This file tests our triple-parser, but it needs access to `pkgs`
+# because `pkgs.gnu-config` is our source of truth for the
+# correctness of parsing.  This file does not test `pkgs`; it tests
+# `lib`.
+#
+# The tests below attempt to verify that parsing followed by
+# unparsing behaves the same way as `gnu-config`, except in the
+# rare and unusual cases where nixpkgs has carefully and
+# deliberately decided to diverge from it.
+#
+# We use `gnu-config` as our source of ground truth not because it
+# is elegant (it isn't) or sensible (it isn't), but because it:
+#
+# 1. has an executable (i.e. unambiguous) specification
+# 2. covers a superset of the systems on which nixpkgs could be useful
+# 3. is *not* under nixpkgs' control (bickering/drama avoidance)
```

Comment:
I don't think this point is worth retaining in the comment.

Reply:
Okay, I've pushed 758723a6e32c which tries to use less-negative language.

I do think that it's worth mentioning this feature.  I consider it to be a positive aspect of adopting some existing taxonomy rather than cooking up our own.